apps/angular/src/app/features/anime/table-view/table-view.component.ts

```diff
@@ -1,32 +1,145 @@
-import { map, Observable } from 'rxjs';
+import { BehaviorSubject, combineLatestWith, debounceTime, map, Observable, startWith, switchMap, tap } from 'rxjs';
 
-import { Component, TrackByFunction } from '@angular/core';
-import { AnimeBase } from '@js-camp/core/models/anime';
+import { FormControl } from '@angular/forms';
+import { PageEvent } from '@angular/material/paginator';
+import { Sort, SortDirection } from '@angular/material/sort';
+import { ChangeDetectionStrategy, Component, TrackByFunction, ViewEncapsulation } from '@angular/core';
 
+import { AnimeBase, Type } from '@js-camp/core/models/anime';
+
+import { goToTop } from '../../../../core/utils/animations';
+import { UrlService } from '../../../../core/services/url.service';
 import { AnimeService } from '../../../../core/services/anime.service';
 
+interface TableSort {
+
+  /** The field by which to sort. */
+  readonly field: string;
+
+  /** The sort order. */
+  readonly direction: SortDirection;
+}
+
+interface FilterItem {
+
+  /** Key of filter. */
+  readonly field: string;
+
+  /** Name of filter. */
+  readonly title: string;
+
+  /** Is a filter selected. */
+  readonly isSelect: boolean;
+}
+
+const DEFAULT_SORT_FIELD = 'title_eng';
+const DEFAULT_SORT_DIRECTION = 'asc';
+const INITIAL_PAGE = 0;
+const INPUT_DEBOUNCE_TIME = 500;
+
 /** Table view component. */
 @Component({
   selector: 'app-table-view',
   templateUrl: './table-view.component.html',
   styleUrls: ['./table-view.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
 })
 export class TableViewComponent {
 
+  /** Total number of records for the current query. */
+  public animeListCount = 0;
+
+  /** Number of records per page. */
+  public pageSize = 5;
+
+  /** All possible type filters. */
+  public filterListByType: FilterItem[] = [];
+
+  /** Value of search input. */
+  public readonly search = new FormControl('');
+
+  /** Filter by type. */
+  public readonly typeFilter = new FormControl<Type[]>([Type.Tv]);
+
+  /** Current sort settings. */
+  public readonly sort$ = new BehaviorSubject<TableSort>({
+    field: DEFAULT_SORT_FIELD,
+    direction: DEFAULT_SORT_DIRECTION,
+  });
+
+  /** Current page number. */
+  public readonly currentPageNumber$ = new BehaviorSubject<number>(0);
+
   /** Table columns names. */
   public readonly displayedColumns: readonly string[] = ['image', 'title-english', 'title-japanese', 'aired-start', 'type', 'status'];
 
   /** Anime list. */
   public readonly animeList$: Observable<readonly AnimeBase[]>;
 
-  public constructor(private readonly animeService: AnimeService) {
-    this.animeList$ = this.getAnimeList();
-  }
+  public constructor(
+    urlService: UrlService,
+    private readonly animeService: AnimeService,
+  ) {
+    this.setFilterListByType();
+    this.setPageSize();
```

Comment:
If I understand correctly, you can't make `page` property `readonly`, because you mutating it in a function, and typescript don't understand that you call the function only in a constructor. To fix it, I suggest refactoring it like so 
```suggestion
    this.pageSize = this.getInitialPageSize();
```